src/kademlia/traversal_algorithm.cpp

```diff
@@ -600,7 +603,14 @@ void traversal_observer::reply(msg const& m)
 			node_id id;
 			std::copy(nodes, nodes + 20, id.begin());
 			nodes += 20;
-			algorithm()->traverse(id, read_v4_endpoint<udp::endpoint>(nodes));
+#if TORRENT_USE_IPV6
```

Comment:
it seems like this snippet could be simplified. by just doing the reading of the endpoint in the if-block and having a single call to algorithm()->traverse(). and to move the IPv6 case to the top (to only have one #if/#endif block)